containers/transcode/transcode.sh

```diff
@@ -21,4 +22,11 @@ aws s3 cp s3://"$STATE_MACHINE_ARTIFACT_BUCKET_NAME"/"$STATE_MACHINE_ARTIFACT_OB
     $STATE_MACHINE_FFMPEG_INPUT_FILE_OPTIONS -i artifact \
     $STATE_MACHINE_FFMPEG_OUTPUT_FILE_OPTIONS -f $STATE_MACHINE_DESTINATION_FORMAT output
 
+# Assume a role that will have access to the S3 destination bucket, and use
+# that roles credentials for the s3 cp call
+role=$(aws sts assume-role --role-arn "$STATE_MACHINE_S3_DESTINATION_WRITER_ROLE" --role-session-name porter_transcode_task)
```

Comment:
I haven't tested this yet, and by bash isn't good enough to know if using a variable inside a `$()` works

Reply:
Yeah, that should work.  Although ... do you want to check for errors here?

```
if ! role=$(aws sts assume-role blah blah blah); then
    exit $?
fi
```

Reply:
Though I suppose if assume-role failed, then `$role` would be an empty string, and your `jq` calls below would fail.  As long as things fail, I'm fine with that!